### DISCUSSION TITLE
A database statement execution error occurred: disk I/O error

### DISCUSSION THREAD

**Discussion by kin0be** [abraunegg/onedrive#1413] [Q&A] [answered]
```
Configuring Global Azure AD Endpoints

A database statement execution error occurred: disk I/O error

Please retry your command with --resync to fix any local database corruption issues.

Segmentation fault (core dumped)
```

I constantly get this after a first successful sync. 

I use sync_list to only download certain folders and files. On first attempt it works, subsequently this is always the error I get when issuing `--synchronize`, `--synchronize --resync` or `--monitor`. Rebooting PC gives me a database locked error. Logging out from onedrive does not work either, I get the above errors. 

I should mention deleting all files in `~/.config/onedrive` and re-auth and re-syncing works, but I mean, something must be happening where the database is constantly locked to a point I have to manually re-auth and resync? What am I doing wrong? Thanks.

**Comment by abraunegg** [comment, marked as answer]
@kin0be 
Regarding this issue:
```
A database statement execution error occurred: disk I/O error
```

This error occurs when:
* You are running the client with `--monitor` and it is running in the background
* Whilst it is running in the background, you attempt *another* sync using `--monitor` or `--synchronize`

The only way to prevent this issue is to not run the client twice at the same time with the same configuration.

**Reply by habruening** [reply]
It could be worth changeing the error message.
`A database statement execution error occurred: disk I/O error. Is there another instance running?`

**Reply by lisarod** [reply]
This happened to me after I started onedrive as a service (systemctl start onedrive). Once I disabled the service (systemctl stop onedrive), the --resync worked.

**Reply by abraunegg** [reply]
> 

Did you install the client from a package on Ubuntu? If yes then this applies:
* Issue [#1274](https://github.com/abraunegg/onedrive/issues/1274)
* https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md
* https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages

Or .. dont use Ubuntu and avoid all the packaging issues all together byusing a better Linux distribution such as Arch, Manjaro, Fedora, Gentoo, SuSE, Solus ...

**Reply by lisarod** [reply]
CentOS 7

**Comment by sandrock** [comment]
Hi,

I have been using the app for 2 years now and this problem occurs every time I want to update my sync_list. As of today:

First I stop the service:
```
> sandrock@firefly-mjr  ~  sudo systemctl stop onedrive@sandrock.service
[sudo] password for sandrock:
sandrock@firefly-mjr  ~  sudo systemctl status onedrive@sandrock.service
○ onedrive@sandrock.service - OneDrive Free Client for sandrock
     Loaded: loaded (/usr/lib/systemd/system/onedrive@.service; enabled; preset: disabled)
     Active: inactive (dead) since Sat 2022-12-10 17:32:11 CET; 5s ago
   Duration: 1d 8h 52min 20.759s
    Process: 1908005 ExecStart=/usr/local/bin/onedrive --monitor --confdir=/home/sandrock/.config/onedrive (code=exited>
   Main PID: 1908005 (code=exited, status=0/SUCCESS)
Dec 10 17:16:29 firefly-mjr onedrive[1908005]: Sync with OneDrive is complete
Dec 10 17:32:11 firefly-mjr onedrive[1908005]: Got termination signal, shutting down db connection
Dec 10 17:32:11 firefly-mjr systemd[1]: Stopping OneDrive Free Client for sandrock...
Dec 10 17:32:11 firefly-mjr systemd[1]: onedrive@sandrock.service: Deactivated successfully.
Dec 10 17:32:11 firefly-mjr systemd[1]: Stopped OneDrive Free Client for sandrock.
Dec 10 17:32:11 firefly-mjr systemd[1]: onedrive@sandrock.service: Consumed 3min 25.089s CPU time.
```
Service is stopped. Now let's change the list...
```
> sandrock@firefly-mjr  ~  vi ~/.config/onedrive/sync_list
```
And resync:
```
> sandrock@firefly-mjr  ~  onedrive --resync --synchronize
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
A database statement execution error occurred: disk I/O error
Please retry your command with --resync to fix any local database corruption issues.
```
Am I stupid or what?

After changing the sync_list again, `--resync` worked.

I don't get how it repairs itself after playing with various commands.

**Reply by abraunegg** [reply]
@sandrock 
> A database statement execution error occurred: disk I/O error

This error occurs because you *still* have an active instance in the background. This was (and is) a problem if you are using:
* Ubuntu
* Ubuntu Universe Packages
* Ubuntu PPA Packages

> Service is stopped. Now let's change the list...

Double check to ensure that there is zero active instances with `sudo ps -aufxw | grep 'onedrive'` ... 

> I don't get how it repairs itself after playing with various commands.

What application version are you using - what is the output of `onedrive --version` ?

For an explanation of the issue - please read: https://github.com/abraunegg/onedrive/issues/1274

**Reply by sandrock** [reply]
Hi, I have news. I was using some AUR package (don't know which) with v2.4.10-19-g6b20478. On Manjaro.

My discovery is: when I do `systemctl stop onedrive@sandrock.service`, the process does not get killed! Hence the errors... You were right about double checking the process is gone.

Now installing AUR `onedrive-abraunegg` (v2.4.23). It works way better now.

**Reply by abraunegg** [reply]
@sandrock 
> I was using some AUR package (don't know which) with v2.4.10-19-g6b20478. On Manjaro.
> ...
> Now installing AUR onedrive-abraunegg (v2.4.23). It works way better now.

This highlights the importance of you using the correct application version. You were running an application version from Feb-Apr 2021 which was 100% obsolete and unsupported. 

At least now you are running the latest available version. Please keep your application current to avoid future issues.
